### PR TITLE
[GEOS-9443] Search and replace abilities for style editor

### PR DESCRIPTION
**Pull request by aaime**
Add find/find next/replace tooling to the style editor, quite useful when editing large styles.

Selezione_999(067)

Selezione_999(068)

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: doc/en/user/source/styling/webadmin/index.rst [hunk 1]
```diff
@@ -140,6 +140,12 @@ The style editor supports line numbering, automatic indentation, and real-time s
      - Redo
    * - .. image:: img/styles_editor_goto.png
      - Go to line
+   * - .. image:: img/styles_editor_find.png
+     - Find in style text (CTRL-F)
+   * - .. image:: img/styles_editor_find_next.png
+     - Find next occurrence in style text (CTRL-G)
+   * - .. image:: img/styles_editor_replace.png
+     - Find and replace in style text (CTRL-SHIFT-F)
    * - .. image:: img/styles_editor_reformat.png
```
Comment (tbarsballe) [inline doc/en/user/source/styling/webadmin/index.rst]: I found the actual operation of the replace function a bit unintuitive to use - I'm used to having "Replace" and "With" together in the same dialog. A doc note indicating that "Enter" lets you set the "With" is probably a good idea.

File: src/main/src/main/java/org/geoserver/GeoserverInitStartupListener.java [hunk 2]
```diff
@@ -415,6 +415,9 @@ public void contextDestroyed(ServletContextEvent sce) {
             if (!relinquishLoggingControl) LogManager.shutdown();
             LogFactory.release(Thread.currentThread().getContextClassLoader());
 
+            org.geotools.util.WeakCollectionCleaner.DEFAULT.exit();
+            org.geotools.referencing.factory.DeferredAuthorityFactory.exit();
```
[the remark this answers is not in the document]
Comment (aaime) [reply src/main/src/main/java/org/geoserver/GeoserverInitStartupListener.java]: Ah sorry, this is, while useful, unrelated. Should back it out and add in a separate commit.

File: src/web/core/src/main/java/org/geoserver/web/wicket/CodeMirrorEditor.html [hunk 3]
```diff
@@ -70,6 +70,15 @@
...
               </a>
+              <a class="button-find icon separator" href="#" title="Find" id="cm_find">
+                <span></span>
+              </a>
+               <a class="button-find-next icon separator" href="#" title="Find next" id="cm_find_next">
+                 <span></span>
+               </a>
+              <a class="button-replace icon separator" href="#" title="Replace" id="cm_replace">
+                 <span></span>
+              </a>
```
[the remark this answers is not in the document]
Comment (aaime) [reply src/web/core/src/main/java/org/geoserver/web/wicket/CodeMirrorEditor.html]: Found out that using "#a" works equally well, the page does not move around. Applied to all buttons.

File: doc/en/user/source/styling/webadmin/index.rst [hunk 4]
```diff
@@ -140,6 +140,12 @@ The style editor supports line numbering, automatic indentation, and real-time s
      - Redo
    * - .. image:: img/styles_editor_goto.png
      - Go to line
+   * - .. image:: img/styles_editor_find.png
+     - Find in style text (CTRL-F)
+   * - .. image:: img/styles_editor_find_next.png
+     - Find next occurrence in style text (CTRL-G)
```
[the remark this answers is not in the document]
Comment (aaime) [reply doc/en/user/source/styling/webadmin/index.rst]: Yep

**Comment by aaime** [comment]
@tbarsballe changes applied following your feedback

**Pull request merged**